plugins/woocommerce/src/Internal/Admin/Logging/Settings.php

```diff
@@ -0,0 +1,450 @@
+		$settings = array(
+			'start'                 => array(
+				'title' => __( 'Logs settings', 'woocommerce' ),
+				'id'    => self::PREFIX . 'settings',
+				'type'  => 'title',
+			),
+			'logging_enabled'       => array(
+				'title'    => __( 'Logger', 'woocommerce' ),
+				'desc'     => __( 'Enable logging', 'woocommerce' ),
+				'id'       => self::PREFIX . 'logging_enabled',
+				'type'     => 'checkbox',
+				'value'    => $this->logging_is_enabled() ? 'yes' : 'no',
+				'default'  => self::DEFAULTS['logging_enabled'] ? 'yes' : 'no',
+				'autoload' => false,
+			),
+			'default_handler'       => array(),
+			'retention_period_days' => array(),
+			'level_threshold'       => array(),
+			'end'                   => array(
+				'id'   => self::PREFIX . 'settings',
+				'type' => 'sectionend',
+			),
+		);
+
+		if ( true === $this->logging_is_enabled() ) {
+			$settings['default_handler']       = $this->get_default_handler_setting_definition();
+			$settings['retention_period_days'] = $this->get_retention_period_days_setting_definition();
+			$settings['level_threshold']       = $this->get_level_threshold_setting_definition();
+		}
+
+		$default_handler = $this->get_default_handler();
+		if ( in_array( $default_handler, array( LogHandlerFileV2::class, WC_Log_Handler_File::class ), true ) ) {
+			$settings += $this->get_filesystem_settings_definitions();
+		} elseif ( WC_Log_Handler_DB::class === $default_handler ) {
+			$settings += $this->get_database_settings_definitions();
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * The definition for the default_handler setting.
+	 *
+	 * @return array
+	 */
+	private function get_default_handler_setting_definition(): array {
+		$handler_options = array(
+			LogHandlerFileV2::class  => __( 'File system (default)', 'woocommerce' ),
+			WC_Log_Handler_DB::class => __( 'Database (not recommended on live sites)', 'woocommerce' ),
+		);
+
+		/**
+		 * Filter the list of logging handlers that can be set as the default handler.
+		 *
+		 * @param array $handler_options An associative array of class_name => description.
+		 *
+		 * @since 8.6.0
+		 */
+		$handler_options = apply_filters( 'woocommerce_logger_handler_options', $handler_options );
+
+		$current_value = $this->get_default_handler();
+		if ( ! array_key_exists( $current_value, $handler_options ) ) {
+			$handler_options[ $current_value ] = $current_value;
+		}
+
+		$desc = array();
+
+		$desc[] = __( 'Note that if this setting is changed, any log entries that have already been recorded will remain stored in their current location, but will not migrate.', 'woocommerce' );
+
+		$hardcoded = ! is_null( Constants::get_constant( 'WC_LOG_HANDLER' ) );
+		if ( $hardcoded ) {
+			$desc[] = sprintf(
+				// translators: %1$s is the name of a code variable. %2$s is the name of a file.
+				__( 'This setting cannot be changed here because it is defined in the %1$s constant, probably in your %2$s file.', 'woocommerce' ),
```

Comment:
we can probably remove the `probably` statement. tech savvy would already know it, but it's not very helpful for other merchants who are anyway going to reach out to their hosts if they want it changed.

Reply:
Fair point. Updated in cd0925da669e0b7213a7ed6e7993587fbab997ff